azure/services/groups/spec.go

```diff
@@ -19,52 +19,93 @@ package groups
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	asoresourcesv1 "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
-	"sigs.k8s.io/cluster-api-provider-azure/azure/converters"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/aso"
 )
 
 // GroupSpec defines the specification for a Resource Group.
 type GroupSpec struct {
 	Name           string
+	Namespace      string
 	Location       string
 	ClusterName    string
 	AdditionalTags infrav1.Tags
+	Owner          metav1.OwnerReference
 }
 
-// ResourceName returns the name of the group.
-func (s *GroupSpec) ResourceName() string {
-	return s.Name
+// ResourceRef implements aso.ResourceSpecGetter.
```

Comment:
I'm slightly inclined not to change these since what these functions do is already defined here and copy-pasting any more details for each type is sure to get out of sync: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7eacff73fbfa1ac2c9a786c7ad290caca110905b/azure/interfaces.go#L137-L148